animareflection/ui
# Storybook interaction & E2E tests

## Description

##### Task link: https://trello.com/c/k877fYDs/142-set-up-visual-regression-testing-eg-to-verify-variants-correctly-applied

- Added [Storybook test runner](https://storybook.js.org/docs/react/writing-tests/test-runner) (powered by Jest & Playwright)
  - Since the test runner is headed (unlike pure Jest which is headless by default), the Storybook server must be running for the tests to execute
  - Tests **without** a [`play` (CSF)](https://storybook.js.org/docs/react/writing-stories/play-function) function serve as render unit tests; a test is automatically created for all stories without a `play` function and executed with the runner
  - Tests **with** a `play` function perform the above render unit testing, **and** execute the `play` function via the runner, which is useful for more complex test use cases such as integration, interaction, visual (regression), and general E2E testing
  - More tests will be added with https://trello.com/c/5kc23iaF/151-add-interaction-tests-for-all-components, this PR simply bootstraps the test runner and establishes base patterns
- Updated `test` CI workflow to use Storybook test runner


Note that future versions of `storybook-test` will automatically run the server if not already running. See https://github.com/storybookjs/test-runner#future-work.

## Test Steps

- Verify tests execute and pass locally (consider caveat above; the Storybook process needs to be running as well as in a traditional E2E setup)
- If any, provide feedback on structure of the tests (if you have better ideas for naming conventions for the `play`-enhanced stories, test structure, etc.)
